pkg/controller/worker/machines.go

```diff
@@ -186,6 +187,11 @@ func (w *workerDelegate) generateMachineConfig() error {
 				MachineConfiguration: genericworkeractuator.ReadMachineConfiguration(pool),
 			})
 
+			//TODO : remove when AWS CSI driver stops using the aws csi topology key
+
+			//adding aws csi driver topology label if its not specified
+			addAwsCsiDriverTopologyLabel(&machineDeploymentIndex, zone, pool.Labels, machineDeployments)
```

Comment:
`zoneIndex` could've been used instead of creating `machineDeploymentIndex`. 

However, I feel that this `addAwsCsiDriverTopologyLabel` can be updated to get rid of passing machineDeployment and index info. You can rather keep it to just `addAwsCsiDriverTopologyLabel(pool.Labels, zone)` and invoke it directly while adding the label [here](https://github.com/gardener/gardener-extension-provider-aws/blob/58b5865d35527675548c2ef860b6e9381aa6ecfb/pkg/controller/worker/machines.go#L184), similar to [this](https://github.com/gardener/gardener-extension-provider-aws/blob/58b5865d35527675548c2ef860b6e9381aa6ecfb/pkg/controller/worker/machines.go#L181). This way you avoid passing the machineDeployment and tracking the index.

Reply:
yes that would be much neater. Thanks ,will do that

Reply:
find the changes to helper function 36619f9